pkg/services/annotations/annotationsimpl/loki/historian_store.go

```diff
@@ -48,28 +70,229 @@ func NewLokiHistorianStore(cfg setting.UnifiedAlertingStateHistorySettings, ft f
 }
 
 func (r *LokiHistorianStore) Get(ctx context.Context, query *annotations.ItemQuery, accessResources *accesscontrol.AccessResources) ([]*annotations.ItemDTO, error) {
-	return []*annotations.ItemDTO{}, nil
+	rule := &ngmodels.AlertRule{}
+	if query.AlertID != 0 {
+		var err error
+		rule, err = getRule(ctx, r.db, query.OrgID, query.AlertID)
+		if err != nil {
+			if errors.Is(err, ErrMissingRule) {
+				return make([]*annotations.ItemDTO, 0), ErrLokiStoreInvalidQuery.Errorf("rule with ID %d does not exist", query.AlertID)
+			}
+			return make([]*annotations.ItemDTO, 0), ErrLokiStoreInternal.Errorf("failed to query rule: %w", err)
+		}
+	}
+
+	logQL, err := historian.BuildLogQuery(buildHistoryQuery(query, accessResources.Dashboards, rule.UID))
+	if err != nil {
+		return make([]*annotations.ItemDTO, 0), ErrLokiStoreInternal.Errorf("failed to build loki query: %w", err)
+	}
+
+	now := time.Now().UTC()
+	if query.To == 0 {
+		query.To = now.UnixMilli()
+	}
+	if query.From == 0 {
+		query.From = now.Add(-defaultQueryRange).UnixMilli()
+	}
```

Comment:
should we validate the query interval after we populated the defaults? For example, when "from" is user specified and "to" is default?

Reply:
I think so, but we'd probably want to keep that behavior consistent between this, the xorm annotation store, and alert state history. That feels like it's outside the scope of this PR.